mlflow/pyfunc/__init__.py

```diff
@@ -946,15 +966,52 @@ def load_model(
 
     _add_code_from_conf_to_system_path(local_path, conf, code_key=CODE)
     data_path = os.path.join(local_path, conf[DATA]) if (DATA in conf) else local_path
-    model_config = _get_overridden_pyfunc_model_config(
-        conf.get(MODEL_CONFIG, None), model_config, _logger
-    )
+
+    if isinstance(model_config, str):
+        model_config = _validate_and_get_model_config_from_file(model_config)
+
+    conf_model_config = conf.get(MODEL_CONFIG, None)
+    if isinstance(conf_model_config, str):
+        conf_model_config_path = os.path.join(local_path, os.path.basename(conf_model_config))
+        conf_model_config = _validate_and_get_model_config_from_file(conf_model_config_path)
+
+    model_config = _get_overridden_pyfunc_model_config(conf_model_config, model_config, _logger)
+    if model_config:
+        conf.update({MODEL_CONFIG: model_config})
 
     try:
-        if model_config:
-            model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path, model_config)
-        else:
+        if "langchain" in conf[MAIN]:
```

Comment:
Why are we checking for langchain in pyfunc code?

Reply:
can we add a comment describing this? or make it so that the langchain case (even w/ model_config) falls into the later code blocks.

Reply:
See https://github.com/mlflow/mlflow/pull/11951/files/670db08a987ce38d81813b061bff469dd33a8689#r1600547555. I think this solves the problem without any special casing.